storage: TestPushTxnHeartbeatTimeout failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/0dba537ae88e495ddf29b4c347b4c30ee99bd046

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestPushTxnHeartbeatTimeout PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=985354&tab=buildLog

```
=== RUN   TestPushTxnHeartbeatTimeout
I181025 05:54:46.763449 24718 storage/store.go:1547  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
--- FAIL: TestPushTxnHeartbeatTimeout (0.02s)
	replica_test.go:5613: 9: expSuccess=false; got pErr <nil>, args={RequestHeader:{Key:"key-9" EndKey:/Min Sequence:0} PusherTxn:"pusher" id=e7955490 key="key-9" rw=false pri=0.00733603 iso=SERIALIZABLE stat=PENDING epo=0 ts=0.000000123,124 orig=0.000000123,124 max=0.000000124,124 wto=false rop=false seq=0 PusheeTxn:{ID:bba4a87f-571b-4a8a-80b8-9d555b6b37d4 Isolation:SERIALIZABLE Key:[107 101 121 45 57] Epoch:0 Timestamp:0.000000123,123 Priority:0 Sequence:1 DeprecatedBatchIndex:0} PushTo:2.000000123,123 Now:2.000000123,123 PushType:PUSH_TIMESTAMP Force:false}, reply=header:<> pushee_txn:<meta:<id:<bba4a87f-571b-4a8a-80b8-9d555b6b37d4> key:"key-9" timestamp:<wall_time:2000000123 logical:124 > sequence:1 > name:"test-9" last_heartbeat:<wall_time:123 logical:127 > orig_timestamp:<wall_time:123 logical:123 > max_timestamp:<wall_time:124 logical:123 > refreshed_timestamp:<> epoch_zero_timestamp:<> > 

```

Reproduced in a little over a minute on `release-2.1`:

```
~ make stress PKG=./storage/ TESTS=TestPushTxnHeartbeatTimeout
...
--- FAIL: TestPushTxnHeartbeatTimeout (0.02s)
	replica_test.go:5613: 4: expSuccess=false; got pErr <nil>, args={RequestHeader:{Key:"key-4" EndKey:/Min Sequence:0} PusherTxn:"pusher" id=1e0eef74 key="key-4" rw=false pri=0.01352606 iso=SERIALIZABLE stat=PENDING epo=0 ts=0.000000123,74 orig=0.000000123,74 max=0.000000124,74 wto=false rop=false seq=0 PusheeTxn:{ID:a364f77e-d29f-4b0a-92de-aca445e76d0b Isolation:SERIALIZABLE Key:[107 101 121 45 52] Epoch:0 Timestamp:0.000000123,73 Priority:0 Sequence:1 DeprecatedBatchIndex:0} PushTo:1.000000123,73 Now:1.000000123,73 PushType:PUSH_ABORT Force:false}, reply=header:<> pushee_txn:<meta:<id:<a364f77e-d29f-4b0a-92de-aca445e76d0b> key:"key-4" timestamp:<wall_time:123 logical:77 > sequence:1 > name:"test-4" status:ABORTED last_heartbeat:<wall_time:123 logical:77 > orig_timestamp:<wall_time:123 logical:73 > max_timestamp:<wall_time:124 logical:73 > refreshed_timestamp:<> epoch_zero_timestamp:<> >
FAIL


ERROR: exit status 1

4205 runs completed, 1 failures, over 1m16s
```

I'm not terribly familiar with this code. Seems a bit worrisome for this to be failing. @andreimatei and @tschottdorf can you also take a look?

Took another crack at reproducing this on master this morning and had success:

```
--- FAIL: TestPushTxnHeartbeatTimeout (0.02s)
	replica_test.go:5711: 7: expSuccess=false; got pErr <nil>, args={RequestHeader:{Key:"key-7" EndKey:/Min Sequence:0} PusherTxn:"pusher" id=846652a6 key="key-7" rw=false pri=0.01264969 iso=SERIALIZABLE stat=PENDING epo=0 ts=0.000000123,108 orig=0.000000123,108 max=0.000000124,108 wto=false rop=false seq=0 PusheeTxn:{ID:091630d6-1485-4166-88c0-8c4b2fe8e545 Isolation:SERIALIZABLE Key:[107 101 121 45 55] Epoch:0 Timestamp:0.000000123,107 Priority:0 Sequence:1 DeprecatedBatchIndex:0} PushTo:2.000000122,107 Now:2.000000122,107 PushType:PUSH_ABORT Force:false}, reply=header:<> pushee_txn:<meta:<id:<091630d6-1485-4166-88c0-8c4b2fe8e545> key:"key-7" timestamp:<wall_time:123 logical:111 > sequence:1 > name:"test-7" status:ABORTED last_heartbeat:<wall_time:123 logical:111 > orig_timestamp:<wall_time:123 logical:107 > max_timestamp:<wall_time:124 logical:107 > refreshed_timestamp:<> epoch_zero_timestamp:<> >
FAIL
```

So whatever is going on, it wasn't fixed on `master` recently.

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -5685,8 +5685,8 @@ func TestPushTxnHeartbeatTimeout(t *testing.T) {
 
 	for i, test := range testCases {
 		key := roachpb.Key(fmt.Sprintf("key-%d", i))
-		pushee := newTransaction(fmt.Sprintf("test-%d", i), key, 1, enginepb.SERIALIZABLE, tc.Clock())
-		pusher := newTransaction("pusher", key, 1, enginepb.SERIALIZABLE, tc.Clock())
+		pushee := newTransaction(fmt.Sprintf("test-%d", i), key, roachpb.MaxUserPriority, enginepb.SERIALIZABLE, tc.Clock())
+		pusher := newTransaction("pusher", key, roachpb.MinUserPriority, enginepb.SERIALIZABLE, tc.Clock())
 
 		// First, establish "start" of existing pushee's txn via BeginTransaction.
 		if test.heartbeat != (hlc.Timestamp{}) {

